### DISCUSSION TITLE
`podman run` bind mounts not created when flags are added to image?

### DISCUSSION THREAD

**Discussion by graham-cs** [containers/podman#24140] [Q&A]
hello,

i'm testing a prometheus container with `-v` bind mounts using:

```
$ podman volume create prometheus-data
$ podman run -d --name prometheus \
    -p 9090:9090 \
    -v /etc/prometheus/prometheus.yml:/etc/prometheus/prometheus.yml \
    -v prometheus-data:/prometheus \
    docker.io/prom/prometheus
```
this seems to work fine and after some testing i ran `$ podman stop prometheus && podman rm prometheus` and then tried the above `podman run` command with the prometheus `--web.enable-admin-api` command line flag appended to the end. the container immediately stopped and the log shows the error `ts=2024-10-02T17:47:26.921Z caller=main.go:549 level=error msg="Error loading config (--config.file=prometheus.yml)" file=/prometheus/prometheus.yml err="open prometheus.yml: no such file or directory"`. additionally appending `--config.file=/etc/prometheus/prometheus.yml` appears to resolve the error, but the `prometheus-data` volume didn't appear to be mounted in the running container.

in other words `podman run`...`docker.io/prom/prometheus` mounts the specified `-v` file and volume without an issue. `podman run`...`docker.io/prom/prometheus --web.enable-admin-api` with the same `-v` options doesn't seem to create the bind mounts?

i tried reading through the man pages for `podman run` to see if this is expected behavior. not sure if i missed something obvious, but posting here in the hopes of clarifying what's happening.

thanks

**Comment by graham-cs** [comment]
looks like i may have been going down the wrong path here, just ran `podman inspect` on the containers created by `podman run`...`docker.io/prom/prometheus` and `podman run`...`docker.io/prom/prometheus --web.enable-admin-api` and the bind mounts are created in both cases. i didn't realize specifying the argument would overwrite the default arguments as opposed to appending to them:

defaults:
```
          "Path": "/bin/prometheus",
          "Args": [

               "--config.file=/etc/prometheus/prometheus.yml",
               "--storage.tsdb.path=/prometheus",
               "--web.console.libraries=/usr/share/prometheus/console_libraries",
               "--web.console.templates=/usr/share/prometheus/consoles"
```

with flags:
```
          "Path": "/bin/prometheus",
          "Args": [
               "--web.enable-admin-api",
               "--config.file=/etc/prometheus/prometheus.yml"
```

pardon the confusion. this was an early test before setting up a compose file which i imagine will sort this all out anyways? just out of curiosity is there an option i missed to keep the default arguments when specifying an additional argument?
